lib/ash_admin/layouts.ex

```diff
@@ -0,0 +1,25 @@
+defmodule AshAdmin.Layouts do
+  @moduledoc false
+  use AshAdmin.Web, :html
+
+  embed_templates "layouts/*"
+
+  def live_socket_path(conn) do
+    [Enum.map(conn.script_name, &["/" | &1]) | conn.private.live_socket_path]
+  end
+
+  def asset_hash("/statics/" <> path) do
+    file =
+      Application.app_dir(:ash_admin, ["priv", "static", path])
```

Comment:
This seems strange. I've not seen something like this in a phoenix application before, how is it normally done for static assets?

Reply:
what exacly it is stange? the way I am reading the asset or reading the assets to get the md5 hash to append to the url?

Let me give you more context, phoenix application normally manages the asset and cache running the task [mix phx.digest](https://hexdocs.pm/phoenix/Mix.Tasks.Phx.Digest.html), the result for that is an asset minified and zipped with this pattern name `asset-#{hash}.#{ext}.gz`
But because ash admin is not an application, its assets are not handle for mix phx.digest, we have to do it runtime or  manual (digest each asset before release any version)

Reply:
I think what feels strange to me is that we're reading a file to hash it to figure out where its minified version is, but we're doing it every time a page load happens. I think it would be better to make this a macro or use a module attribute to store the hash of it at compile time. You can use `@external_dependency` to cause external files to trigger recompiles.